Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/CertificateStateHelper.kt

```diff
@@ -137,10 +137,22 @@ fun PersonOverviewItemBinding.setUIState(
     val statusBadgeText = item.admissionBadgeText
     qrCodeCard.apply {
         loadQrImage(firstCertificate.cwaCertificate)
-        statusText.isVisible = statusBadgeText.isNotEmpty()
-        statusBadge.isVisible = statusBadgeText.isNotEmpty()
-        if (statusBadgeText.isNotEmpty()) {
-            statusBadge.text = statusBadgeText
+        if (item.hasMaskState) {
+            setMaskBadge(maskBadge, item, color)
+        }
+        statusBadge.setBackgroundResource(color.admissionBadgeBg)
+        statusBadge.text = statusBadgeText
+        if (statusBadgeText.isEmpty() && !item.hasMaskState) {
+            statusBadge.visibility = View.GONE
+        } else if (statusBadgeText.isEmpty() && item.hasMaskState) {
+            statusBadge.visibility = View.INVISIBLE
+        } else if (statusBadgeText.isNotEmpty()) {
+            statusBadge.visibility = View.VISIBLE
+        }
```

Comment:
looks like a good candidate of `when` instead

Reply:
Changed to when